internal/provider/resource_tfe_project_oauth_client_.go

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceTFEProjectOAuthClient() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceTFEProjectOauthClientCreate,
+		Read:   resourceTFEProjectOauthClientRead,
+		Delete: resourceTFEProjectOauthClientDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceTFEProjectOauthClientImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"oauth_client_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceTFEProjectOauthClientCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(ConfiguredClient)
+
+	oauthClientID := d.Get("oauth_client_id").(string)
+	projectID := d.Get("project_id").(string)
+
+	oauthClientAddProjectsOptions := tfe.OAuthClientAddProjectsOptions{}
+	oauthClientAddProjectsOptions.Projects = append(oauthClientAddProjectsOptions.Projects, &tfe.Project{ID: projectID})
```

Comment:
Great work! The `Project` struct has two fields `ID` and `Name`.  I'm wondering if it's a helpful experience to also capture `Name`? This is non-blocking, just curious.